### ISSUE TITLE
error with podman 1.9.0 and cgroup1

### ISSUE THREAD

**Issue by CameronNemo** [containers/libpod#5874]
/kind bug

**Description**

podman 1.9.0 fails to run a rootless container with cgroup1, while 1.8.2 is able to run.

**Steps to reproduce the issue:**

1. Configure a fairly standard cgroup1 setup. Mount output: http://ix.io/2iuO
1. Invoke any run command, e.g. `podman --log-level DEBUG run --rm -it docker://cameronnemo/voidlinux-interactive:latest ash`

**Describe the results you received:**

* Debug log with 1.9.0: http://ix.io/2iuM
* Debug log with 1.8.2: http://ix.io/2iuN

**Describe the results you expected:**

A successful container run, like with 1.8.2.

**Additional information you deem important (e.g. issue happens only occasionally):**

Happens consistently.

**Output of `podman version`:**

1.9.0:

```
Version:            1.9.0
RemoteAPI Version:  1
Go Version:         go1.14.2
OS/Arch:            linux/amd64
```

1.8.2:

```
Version:            1.8.2
RemoteAPI Version:  1
Go Version:         go1.14
OS/Arch:            linux/amd64
```

**Output of `podman info --debug`:**

1.9.0:

```yaml
debug:
  compiler: gc
  gitCommit: ""
  goVersion: go1.14.2
  podmanVersion: 1.9.0
host:
  arch: amd64
  buildahVersion: 1.14.8
  cgroupVersion: v1
  conmon:
    package: Unknown
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.15, commit: unknown'
  cpus: 4
  distribution:
    distribution: '"void"'
    version: unknown
  eventLogger: file
  hostname: ceci
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 2000000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 2000000
      size: 65536
  kernel: 5.4.31_1
  memFree: 503386112
  memTotal: 8106188800
  ociRuntime:
    name: runc
    package: Unknown
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.1-dev'
  os: linux
  rootless: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: Unknown
    version: |-
      slirp4netns version 1.0.0
      commit: a3be729152a33e692cd28b52f664defbf2e7810a
      libslirp: 4.2.0
  swapFree: 2046029824
  swapTotal: 2047995904
  uptime: 176h 46m 36.14s (Approximately 7.33 days)
registries:
  search:
  - docker.io
store:
  configFile: /home/cameronnemo/.config/containers/storage.conf
  containerStore:
    number: 2
    paused: 0
    running: 0
    stopped: 2
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: Unknown
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 0.7.8
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  graphRoot: /home/cameronnemo/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 12
  runRoot: /run/user/1000
  volumePath: /home/cameronnemo/.local/share/containers/storage/volumes
```

1.8.2:

```yaml
debug:
  compiler: gc
  git commit: ""
  go version: go1.14
  podman version: 1.8.2
host:
  BuildahVersion: 1.14.3
  CgroupVersion: v1
  Conmon:
    package: Unknown
    path: /usr/libexec/podman/conmon
    version: 'conmon version 2.0.15, commit: unknown'
  Distribution:
    distribution: '"void"'
    version: unknown
  IDMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 2000000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 2000000
      size: 65536
  MemFree: 496386048
  MemTotal: 8106188800
  OCIRuntime:
    name: runc
    package: Unknown
    path: /usr/bin/runc
    version: 'runc version spec: 1.0.1-dev'
  SwapFree: 2046029824
  SwapTotal: 2047995904
  arch: amd64
  cpus: 4
  eventlogger: file
  hostname: ceci
  kernel: 5.4.31_1
  os: linux
  rootless: true
  slirp4netns:
    Executable: /usr/bin/slirp4netns
    Package: Unknown
    Version: |-
      slirp4netns version 1.0.0
      commit: a3be729152a33e692cd28b52f664defbf2e7810a
      libslirp: 4.2.0
  uptime: 176h 48m 16.25s (Approximately 7.33 days)
registries:
  search:
  - docker.io
store:
  ConfigFile: /home/cameronnemo/.config/containers/storage.conf
  ContainerStore:
    number: 2
  GraphDriverName: overlay
  GraphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: Unknown
      Version: |-
        fusermount3 version: 3.9.1
        fuse-overlayfs: version 0.7.8
        FUSE library version 3.9.1
        using FUSE kernel interface version 7.31
  GraphRoot: /home/cameronnemo/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  ImageStore:
    number: 12
  RunRoot: /run/user/1000
  VolumePath: /home/cameronnemo/.local/share/containers/storage/volumes
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

1.9.0 (locally built and installed):

```
architecture: x86_64
filename-sha256: bc97aabc549fbc6451eb772dec196a1a0d5aa118a478a91e8e45eecf9d07942f
filename-size: 20MB
homepage: https://podman.io/
install-date: 2020-04-18 02:02 PDT
installed_size: 55MB
license: Apache-2.0
maintainer: Cameron Nemo <cnemo@tutanota.com>
metafile-sha256: 61a97cc0fe02eb503bd248b9bde34de6cf57602baea3ec3dc62e44fc18776397
pkgname: podman
pkgver: podman-1.9.0_1
repository: /home/cameronnemo/docs/src/vpkg/hostdir/binpkgs/podman-1.9.0
run_depends:
	runc>=0
	conmon>=0
	cni-plugins>=0
	slirp4netns>=0
	containers.image>=0
	glibc>=2.29_1
	libgpgme>=1.12.0_2
	libassuan>=2.0.1_1
	libgpg-error>=1.6_1
	libseccomp>=2.0.0_1
	device-mapper>=2.02.110_1
shlib-requires:
	libpthread.so.0
	libgpgme.so.11
	libassuan.so.0
	libgpg-error.so.0
	libseccomp.so.2
	librt.so.1
	libdevmapper.so.1.02
	libc.so.6
short_desc: Simple management tool for containers and images
state: installed
```

1.8.2 (built by repos:

```
architecture: x86_64
filename-sha256: afd84e7980ed2610e232f3d2f14e60aa210656e77b4e705c46f9d861cdbce762
filename-size: 20MB
homepage: https://podman.io/
install-date: 2020-04-18 02:05 PDT
installed_size: 54MB
license: Apache-2.0
maintainer: Cameron Nemo <camerontnorman@gmail.com>
metafile-sha256: 3df90315902b0baac657941b27d7e76cfd493c9391c734fe10253eb3e2c207a8
pkgname: podman
pkgver: podman-1.8.2_2
repository: https://alpha.de.repo.voidlinux.org/current
run_depends:
	runc>=0
	conmon>=0
	cni-plugins>=0
	slirp4netns>=0
	containers.image>=0
	glibc>=2.29_1
	libgpgme>=1.12.0_2
	libassuan>=2.0.1_1
	libgpg-error>=1.6_1
	libseccomp>=2.0.0_1
	device-mapper>=2.02.110_1
shlib-requires:
	libpthread.so.0
	libgpgme.so.11
	libassuan.so.0
	libgpg-error.so.0
	libseccomp.so.2
	librt.so.1
	libdevmapper.so.1.02
	libc.so.6
short_desc: Simple management tool for containers and images
source-revisions: podman:34432dd49f
state: installed
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

Physical.

**Comment by rhatdan** [comment]
podman --log-level DEBUG run --rm -it docker://cameronnemo/voidlinux-interactive:latest ash 
Works for me on cgroups V1
$ rpm -q podman
podman-1.9.0-1.fc32.x86_64

@giuseppe Ideas?

**Comment by CameronNemo** [comment]
One difference may be the management method -- on Void we do not have systemd so it is using the cgroupsfs manager.

**Comment by giuseppe** [comment]
are you using exactly the same runc version for both?

Could you share the `config.json` file that is generated (you can get the path from the debug output: `Created OCI spec for container $CTR_ID at $PATH`)?

**Comment by s-hamann** [comment]
I believe I have the same issue on Gentoo without Systemd.
The following line in `dmesg` seems informative:
```
cgroup1: Unknown subsys name 'openrc'
```
With a hybrid cgroup setup (probably with legacy v1 as well) I have `/sys/fs/cgroup/openrc`, mounted as
```
openrc on /sys/fs/cgroup/openrc type cgroup (rw,nosuid,nodev,noexec,relatime,release_agent=/lib/rc/sh/cgroup-release-agent.sh,name=openrc)
```

**Comment by mheon** [comment]
We do not support hybrid cgroups in Podman - that's likely the issue there

**Comment by s-hamann** [comment]
> We do not support hybrid cgroups in Podman

I am aware of that, but it works just fine with Podman 1.8.2. I believe it is detected as cgroup v1, and the cgroup v2 mount is ignored, right?
So, my guess is that a cgroup v1 setup (with openrc) would have the same issue. I did not test that, though.
I did test cgroup v2, but could not get Podman to run either (due to a different issue, probably #5903).

**Comment by CameronNemo** [comment]
Same runc (same everything) for each version. I am not using the unified hierarchy.

[1.8.2 config.json](http://ix.io/2iYG)

[1.9.0 config.json](http://ix.io/2iYF)

**Comment by ocroz** [comment]
Meanwhile this issue is still opened, how can we downgrade podman to 1.8.2 ?
Following the [install](https://podman.io/getting-started/installation.html) instructions, the [Kubic project](https://build.opensuse.org/project/show/devel:kubic:libcontainers:stable) seems only to provide the latest version 1.9.0.

**Comment by CameronNemo** [comment]
This appears to be fixed with the 1.9.2 release.
